clients/client-direct/src/index.ts

```diff
@@ -334,25 +386,118 @@ export class DirectClient {
 					createdAt: Date.now(),
 				};
 
+				// Log memory storage event
+				instrumentation.logEvent({
+					stage: "Chat",
+					subStage: "Memory",
+					event: "memory_storage_started",
+					data: {
+						messageId,
+						sessionId,
+						agentId: runtime.agentId,
+						roomId,
+						userId,
+						timestamp: Date.now(),
+					},
+				});
+
 				await runtime.messageManager.addEmbeddingToMemory(memory);
 				await runtime.messageManager.createMemory(memory);
 
+				// Log memory storage complete event
+				instrumentation.logEvent({
+					stage: "Chat",
+					subStage: "Memory",
+					event: "memory_storage_completed",
+					data: {
+						messageId,
+						sessionId,
+						agentId: runtime.agentId,
+						roomId,
+						userId,
+						timestamp: Date.now(),
+					},
+				});
+
+				// Log state composition started
+				instrumentation.logEvent({
+					stage: "Chat",
+					subStage: "State",
+					event: "state_composition_started",
+					data: {
+						messageId,
+						sessionId,
+						agentId: runtime.agentId,
+						roomId,
+						userId,
+						timestamp: Date.now(),
+					},
+				});
+
 				let state = await runtime.composeState(userMessage, {
 					agentName: runtime.character.name,
 				});
 
-				const context = composeContext({
-					state,
-					template: messageHandlerTemplate,
+				// Log response generation started
+				instrumentation.logEvent({
+					stage: "Chat",
+					subStage: "Generation",
+					event: "response_generation_started",
+					data: {
+						messageId,
+						sessionId,
+						agentId: runtime.agentId,
+						roomId,
+						userId,
+						modelClass: ModelClass.LARGE,
+						timestamp: Date.now(),
+					},
 				});
 
+				const generationStartTime = Date.now();
 				const response = await generateMessageResponse({
 					runtime: runtime,
-					context,
+					context: composeContext({
+						state,
+						template: messageHandlerTemplate,
+					}),
 					modelClass: ModelClass.LARGE,
 				});
+				const generationEndTime = Date.now();
+
+				// Log response generation completed
+				instrumentation.logEvent({
+					stage: "Chat",
+					subStage: "Generation",
+					event: "response_generation_completed",
+					data: {
+						messageId,
+						sessionId,
+						agentId: runtime.agentId,
+						roomId,
+						userId,
+						generationTime: generationEndTime - generationStartTime,
+						responseLength: response ? response.text.length : 0,
+						timestamp: generationEndTime,
+					},
+				});
```

Comment:
You need to actually store the response

Reply:
Great work thanks for doing this! Remember we need to store the value of composeContext the value of composeState and the value of the action taken or not taken you can do this in a follow up PR but remember how this is being used.